python-package/lightgbm/callback.py

```diff
@@ -438,6 +437,18 @@ def __call__(self, env: CallbackEnv) -> None:
             self._final_iteration_check(env, eval_name_splitted, i)
 
 
+def _should_enable_early_stopping(stopping_rounds: Any) -> bool:
+    """Check if early stopping should be activated.
+
+    This function will evaluate to True if the early stopping callback should be
+    activated (i.e. stopping_rounds > 0).  It also provides an informative error if the
+    type is not int.
+    """
+    if not isinstance(stopping_rounds, int):
+        raise TypeError(f"early_stopping_round should be an integer. Got {type(stopping_rounds)}")
```

Comment:
```suggestion
        raise TypeError(f"early_stopping_round should be an integer. Got {type(stopping_rounds).__name__}")
```

Sorry for not mentioning this in https://github.com/microsoft/LightGBM/pull/6406#discussion_r1550360478. Could you please just print the name instead?

```python
type("abc")
# <class 'str'>

type("abc").__name__
# 'str'
```

I think that's a bit easier to read.

We use that pattern in several other places in the library, I'm fairly sure it's safe.

For example:

https://github.com/microsoft/LightGBM/blob/28536a0a5d0c46598fbc930a63cec72399a969e4/python-package/lightgbm/engine.py#L164-L167

https://github.com/microsoft/LightGBM/blob/28536a0a5d0c46598fbc930a63cec72399a969e4/python-package/lightgbm/basic.py#L2194